sharding/shard.go

```diff
@@ -0,0 +1,208 @@
+package sharding
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// Shard base struct.
+type Shard struct {
+	shardDB *shardKV
+	shardID *big.Int
+}
+
+// MakeShard creates an instance of a Shard struct given a shardID.
+func MakeShard(shardID *big.Int) *Shard {
+	// Swappable - can be makeShardLevelDB, makeShardSparseTrie, etc.
+	shardDB := makeShardKV()
+
+	return &Shard{
+		shardID: shardID,
+		shardDB: shardDB,
+	}
+}
+
+// ShardID gets the shard's identifier.
+func (s *Shard) ShardID() *big.Int {
+	return s.shardID
+}
+
+// ValidateShardID checks if header belongs to shard.
+func (s *Shard) ValidateShardID(h *CollationHeader) error {
+	if s.ShardID().Cmp(h.ShardID()) != 0 {
+		return fmt.Errorf("Error: Collation Does Not Belong to Shard %d but Instead Has ShardID %d", h.ShardID(), s.ShardID())
```

Comment:
Please do not capitalize first word.
Please use regular case words.
Please do not write “Error:” this is already an error and the logger might append “Error:” anyway, but it’s not up to this method to decide. 

Please fix everywhere in this file